include/deal.II/base/symmetric_tensor.h

```diff
@@ -1538,6 +1538,20 @@ namespace internal
     template <typename Number>
     struct Inverse<4, 3, Number>
     {
+      /**
+       * Replacement of std::swap for numbers that
+       * also runs on device.
+       */
+      DEAL_II_HOST_DEVICE
+      template <class Number>
+      void
+      swap(Number &x, Number &y)
+      {
+        Number tmp = x;
+        x          = y;
+        y          = tmp;
+      }
```

Comment:
https://github.com/kokkos/kokkos/pull/6697 Kokkos (4.3.00) adds `Kokkos::kokkos_swap` which you could try using here.
```suggestion
      DEAL_II_HOST_DEVICE
      void
      swap(Number &x, Number &y)
      {
#if DEAL_II_KOKKOS_VERSION_GTE(4 ,3, 0)
        return Kokkos::kokkosswap(x,y);
#else
        Number tmp = x;
        x          = y;
        y          = tmp;
#endif
      }
```